celery_progress/websockets/tasks.py

```diff
@@ -1,12 +1,17 @@
 from celery.signals import task_postrun
 
-from .backend import channel_layer, WebSocketProgressRecorder
+from .backend import WebSocketProgressRecorder
 
 
-@task_postrun.connect
+@task_postrun.connect(retry=True)
 def task_postrun_handler(task_id, **kwargs):
     """Runs after a task has finished. This will be used to push a websocket update for completed events.
 
-    If the websockets version of this package is not installed, this will do nothing."""
-    if channel_layer:
-        WebSocketProgressRecorder.push_update(task_id)
+    If the websockets version of this package is not installed, this will fail silently."""
```

Comment:
is this true? from just reading it I would have expected the import statement to fail.

Reply:
Improper raising of error + AttributeException handled will cause this to fail silently if conditions are not met.